packages/kbn-config-schema/src/types/object_type.ts

```diff
@@ -33,25 +33,32 @@ export type ObjectResultType<P extends Props> = Readonly<{ [K in keyof P]: TypeO
 export type ObjectTypeOptions<P extends Props = any> = TypeOptions<
   { [K in keyof P]: TypeOf<P[K]> }
 > & {
+  /** Should uknown keys not be defined in the schema be allowed. Defaults to `false` */
   allowUnknowns?: boolean;
+  /** Should the `defaultValue` for keys be used if no value is specified. Defaults to `true` */
+  applyDefaults?: boolean;
 };
 
 export class ObjectType<P extends Props = any> extends Type<ObjectResultType<P>> {
   private props: Record<string, AnySchema>;
 
-  constructor(props: P, options: ObjectTypeOptions<P> = {}) {
+  constructor(
+    props: P,
+    { allowUnknowns = false, applyDefaults = true, ...typeOptions }: ObjectTypeOptions<P> = {}
```

Comment:
do you use `applyDefaults: false` anywhere?

Reply:
This is mostly to address the situation outlined in your comment above. My first inclination was to make `applyDefaults` false by default, because I believe that matches the expected semantics more closely, however when looking at usages, we almost always want to apply the defaults, so I thought it made more sense to make that the default behavior.

Reply:
> do you use applyDefaults: false anywhere?

I actually need it for https://github.com/elastic/kibana/pull/56734, where a parameter is a `maybe object with requires props`, which currently fails validation when the query parameter is empty

Reply:
I think that could be supported without this option? For example, this would fail validation

```ts
schema.maybe(schema.object({
  key1: schema.string(),
  key2: schema.string(),
}).validate({ key1: 'something' });
```

It fails if the input does not include all the keys that do not have defaults. Would that fit your situation? If so, I may remove this option (same as always `true` -- the current default)

Reply:
Not sure if I understand correctly.

My example that currently fails is the following
```typexcript
schema.maybe(schema.object({
  key1: schema.string(),
  key2: schema.string(),
}).validate(undefined); // this throws as it tries to apply defaults for key1/key2
```

and I guess I would need

```typexcript
schema.maybe(schema.object({
  key1: schema.string(),
  key2: schema.string(),
}, {applyDefaults: false }).validate(undefined);
```

to works properly.

Reply:
Nevermind, the first point of your PR is what is fixing this unexpected behavior. Not sure about the `applyDefaults` usage then.

Reply:
let's introduce `applyDefaults` when it's required then. @pgayvallet can add it back if it's necessary for https://github.com/elastic/kibana/pull/56734

Reply:
Done.